pkg/diag/errors.go

```diff
@@ -58,3 +58,7 @@ func GetDuplicateResourceAliasError(urn resource.URN) *Diag {
 		"Duplicate resource alias '%v' applied to resource with URN '%v' conflicting with resource with URN '%v'",
 	)
 }
+
+func GetResourceToRefreshCouldNotBeFoundError() *Diag {
+	return newError("", 2010, "Resource to refresh (%v) could not be found in the stack.")
```

Comment:
@lukehoban for wordsmithing ideas.  We could consider looking to see if there's no `$` in the urn and giving an explicit message about that.